client/app/lib/onboarding.coffee

```diff
@@ -0,0 +1,111 @@
+# Local class Step
+class Step
+    # Retrieves properties from config Step plain object
+    constructor: (step) ->
+        if step
+            ['name', 'route', 'view'].forEach (property) =>
+                @[property] = step[property]
+
+
+    # Record handlers for 'validated' internal pseudo-event
+    onValidated: (callback) ->
+        throw new Error 'Callback parameter should be a function' \
+            unless typeof callback is 'function'
+        @validatedHandlers = @validatedHandlers or []
+        @validatedHandlers.push callback
+
+
+    # Trigger 'validated' pseudo-event
+    triggerValidated: () ->
+        if @validatedHandlers
```

Comment:
I fixed as required but I changed my mind about this one :
* First, if we declare it as a class property it will be a global static variable, shared by  all instances
* Second, another way is to declare `@validateHandlers` as an empty array in `initialize`, but I am not comfortable with the way of always declaring an empty array for every instance. I prefer to instanciate it on demand in `onStepValidated` and check existence in `triggerValidated`. Even if in fact and in our code, Step will always have handlers in `validatedHandlers`. So if you don't mind I will let this part as it.